presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -176,6 +182,17 @@ protected InternalResourceGroup(
         this.staticResourceGroup = staticResourceGroup;
         this.additionalRuntimeInfo = requireNonNull(additionalRuntimeInfo, "additionalRuntimeInfo is null");
         this.shouldWaitForResourceManagerUpdate = requireNonNull(shouldWaitForResourceManagerUpdate, "shouldWaitForResourceManagerUpdate is null");
+        initQueryLevelLimits();
+    }
+
+    @GuardedBy("root")
```

Comment:
this doesn't make sense here. It is normally added to variables to tell the callsites wherever this variable is used to ensure that the appropriate synchronized block is used.